flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/FlinkKinesisProducer.java

```diff
@@ -160,12 +168,13 @@ public void setCustomPartitioner(KinesisPartitioner<OUT> partitioner) {
 	public void open(Configuration parameters) throws Exception {
 		super.open(parameters);
 
-		KinesisProducerConfiguration config = new KinesisProducerConfiguration();
-		config.setRegion(this.region);
-		config.setCredentialsProvider(new StaticCredentialsProvider(new BasicAWSCredentials(this.accessKey, this.secretKey)));
+		KinesisProducerConfiguration producerConfig = new KinesisProducerConfiguration();
+
+		producerConfig.setRegion(configProps.getProperty(KinesisConfigConstants.CONFIG_AWS_REGION));
+		producerConfig.setCredentialsProvider(AWSUtil.getCredentialsProvider(configProps));
 		//config.setCollectionMaxCount(1);
 		//config.setAggregationMaxCount(1);
```

Comment:
I agree that using the KPL is in general a good idea.
The problem is that the license of the KPL (the Amazon Software License) is not compatible with the Apache Software License, therefore, projects at Apache can not depend on such code.
We will not release the Flink kinesis connector with the next Flink release for that reason (unless we somehow fix this issue).

Are you working at Amazon? If so, can you contact me at robert (at) data-artisans.com ?